### PR TITLE
refactor: init command and action now covering 100% using vitest

### PR DESCRIPTION
**Pull request by epsjunior**
### Key Changes
- Replaced Jest with Vitest in all test files.
- Updated test configuration and scripts.
- Resolved minor discrepancies for compatibility.

### CODE REVIEW

**Comment by epsjunior** [comment]
Evidences:

init coverage before (Main):
Screenshot 2024-11-05 at 08 25 44

init coverage now:
Screenshot 2024-11-05 at 08 25 44

File: tests/actions/init.test.ts [hunk 1]
```diff
@@ -1,255 +1,124 @@
...
-    await initAction(defaultActionOptions, simulatorService);
-
-    // Then
-    expect(error).toHaveBeenCalledTimes(1);
-    expect(error).toHaveBeenCalledWith(
-      `Docker version ${mockVersionNumber} or higher is required. Please update Docker and try again.\nNode version ${mockVersionNumber} or higher is required. Please update Node and try again.\n`,
-    );
-  });
-
-  test("if only docker version is too low, then the execution fails", async () => {
```
Comment (cristiam86) [inline tests/actions/init.test.ts:126]: @epsjunior why do you want to remove these ones?

**Review by cristiam86 — APPROVED** [review]

**Review by cristiam86 — CHANGES_REQUESTED** [review]
Just minor comments. 
Please adjust the PR title to conventional commits

File: tests/actions/init.test.ts (hunk 1 above)
Comment (epsjunior) [reply tests/actions/init.test.ts:126]: @cristiam86  the if statements were already covered by other tests, but you're right, it's better to create specific tests for these cases to make the coverage clearer and more targeted.

Evidences:
https://github.com/yeagerai/genlayer-cli/pull/115/commits/fa6337d15fc1afed0dbc28fd37250907f90e26ca
Screenshot 2024-11-06 at 06 17 56

**Review by cristiam86 — APPROVED** [review]

**Pull request merged**